Parsely/wp-parsely
# Adding GraphQL support

## Description

This PR adds GraphQL support to output metadata. We are doing so by extending the WPGraphQL plugin (thus, it is a requirement that it is installed). If it's not installed, wp-parsely will continue to work normally but without GraphQL support.

Since GraphQL is strongly typed, we have had to define the types of our metadata. Therefore, in this change we're registering a number of custom types that support our metadata (see screenshots). Then, we are registering the `parsely` field in all fields that should be tracked (added by the user).

In this implementation, we've also added a few filters so user can build on top of our default support.

## Motivation and Context

Closes #190 

## How Has This Been Tested?

1. Install WPGraphQL plugin.
2. Make sure wp-parsely is enabled and with an API key.
3. Perform queries on GraphiQL in wp-admin (see screenshots for examples).

## Screenshots (if appropriate)

<img width="406" alt="Screen Shot 2022-03-09 at 12 15 20 PM" src="https://user-images.githubusercontent.com/7188409/157431042-4a33828c-ddf4-42fb-8472-ddab72d9b062.png">

<img width="389" alt="Screen Shot 2022-03-09 at 12 18 59 PM" src="https://user-images.githubusercontent.com/7188409/157431552-11512dc7-d189-4985-8f7d-135f950352d5.png">

<img width="1656" alt="Screen Shot 2022-03-09 at 12 18 19 PM" src="https://user-images.githubusercontent.com/7188409/157431455-36e74e1c-f86e-491d-a46d-afec53ca774d.png">
